src/vos/vos_common.c

```diff
@@ -157,6 +157,7 @@ vos_tx_publish(struct dtx_handle *dth, bool publish)
 		if (rc && publish)
 			return rc;
 	}
+	dth->dth_rsrvd_cnt = 0;
```

Comment:
Sorry, I am some confused. vos_tx_publish() will be triggered by vos_tx_end() only when the last modification done or some internal operation failed. Under such cases, the DTX handle will be freed after vos_tx_publish(), then both the "dth_rsrvd_cnt" and "dth_deferred_cnt" will disappear. Why need reset them as zero inside vos_tx_publish()?

On the other hand, the patch does not match the commit message.

Reply:
I am also confused by what the bug is and what you are trying to fix.  The ticket is generic so unhelpful and the description doesn't match the code.  Can you elaborate?

Reply:
oh, during OSA, if there are multiple VOSs over the same target, and since it does not use compound RPC for the moment, so the dth might be executed again after vos_tx_end().  i.e. the dth might not be freed in this case, instead it will be reused later.

TBH, I thought this rsrvd_cnt and deferred_cnt should be moved to sub_dtx, since it is totally for VOS, instead of leader dth.

Anyway, set it to 0 after publishing the rsvd is always safer.

Reply:
probably would be better to resolve this with DAOS-5640 to cause less confusion.